src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -198,10 +198,37 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
                 return (int)ExitCode.ErrorInvalidARMTemplate;
             }
 
-            var analysisResult = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            ExitCode exitCode;
+            if (parametersFilePath != null)
+            {
+                exitCode = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            }
+            else
+            {
+                // Check if a parameters.json file is present according to naming standards here https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files#file-name, and if so use it as the parametersFile input
+                var parametersFiles = FindParameterFileForTemplate(templateFilePath);
+                List<ExitCode> exitCodes = new List<ExitCode>();
+
+                if (parametersFiles.Count() > 0)
+                {
+                    foreach (FileInfo parametersFile in parametersFiles)
+                    {
+                        exitCodes.Add(AnalyzeTemplate(templateFilePath, parametersFile));
+                    }
+
+                    exitCode = AnalyzeExitCodes(exitCodes);
+                }
+                else
+                {
+                    exitCode = AnalyzeTemplate(templateFilePath, null);
+                }
+
+                int numOfFilesAnalyzed = exitCodes.Where(x => x == ExitCode.Success || x == ExitCode.Violation).Count();
+                Console.WriteLine(Environment.NewLine + $"Analyzed the template with {numOfFilesAnalyzed} parameter {(numOfFilesAnalyzed == 1 ? "file" : "files")}.");
```

Comment:
Now that I see the output examples you shared I'm thinking this line is kind of unnecessary, the user will already see that multiple parameters files were used, can we remove it?

Reply:
Sure, removed in [`12a325b` (#297)](https://github.com/Azure/template-analyzer/pull/297/commits/12a325b393d6dfd29a1a424a299b00206d4d1d79)

Reply:
Also the previous line, right?

Reply:
Oops you're right, my bad, it's getting late (at least in my timezone 😉), fixed in [`e040c73` (#297)](https://github.com/Azure/template-analyzer/pull/297/commits/e040c73f12817287c5b07aa77d96fc912e180111)